crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -756,357 +654,23 @@ impl Actual {
-                                span: DUMMY_SP,
-                                decorators: Default::default(),
-                                pat: pat.clone(),
-                            }),
-                            Pat::Array(..) | Pat::Object(..) => Some(Param {
-                                span: DUMMY_SP,
-                                decorators: Default::default(),
-                                pat: Pat::Ident(private_ident!("_").into()),
-                            }),
-                            _ => None,
-                        })
-                        .collect(),
-                    body: Some(BlockStmt {
-                        span: DUMMY_SP,
-                        stmts: vec![apply],
-                    }),
-                    decorators: Default::default(),
-                    type_params: Default::default(),
-                    return_type: Default::default(),
-                };
-
-                let return_function = Stmt::Return(ReturnStmt {
-                    span: DUMMY_SP,
-                    arg: Some(Box::new(
-                        FnExpr {
-                            function,
-                            ident: binding_ident,
-                        }
-                        .into(),
-                    )),
-                });
-
-                let block_stmt = BlockStmt {
-                    span: DUMMY_SP,
-                    stmts: vec![ref_stmt, return_function],
-                };
-
-                let function = Function {
-                    span: *span,
-                    body: Some(block_stmt),
-                    params: Default::default(),
-                    is_generator: false,
-                    is_async: false,
-                    decorators: Default::default(),
-                    return_type: Default::default(),
-                    type_params: Default::default(),
-                };
-
-                *expr = Expr::Call(CallExpr {
-                    span: DUMMY_SP,
-                    callee: Expr::Fn(FnExpr {
-                        ident: None,
-                        function,
-                    })
-                    .as_callee(),
-                    args: vec![],
-                    type_args: Default::default(),
-                });
+                *expr = wrapper.into();
             }
 
-            Expr::Fn(
-                fn_expr @ FnExpr {
-                    function:
-                        Function {
-                            is_async: true,
-                            body: Some(..),
-                            ..
-                        },
-                    ..
-                },
-            ) => {
-                self.visit_fn(
-                    fn_expr.ident.clone(),
-                    binding_ident,
-                    &mut fn_expr.function,
-                    false,
-                );
+            Expr::Fn(fn_expr) if fn_expr.function.is_async => {
```

Comment:
Same here, `@ FnExpr{ function: Function { is_async: true, .. }, .. }` would be better.